internal/collectors/collectors.go

```diff
@@ -0,0 +1,53 @@
+/*
+How to use the collector framework:
+In the business logic, we would handle the mandatory/optional metrics
+For hw_infos : we can call this framework on mandatory collectors and send the data to scc
+For tahoe : we can call this with mandatory collectors. And then specify option collectors too.
+
+Example:
+
+	var MandatoryCollectors = []collectors.Collector{
+		collectors.CPU{},
+		collectors.Hostname{},
+		collectors.Architecture{},
+		collectors.Memory{},
+		collectors.UUID{},
+	}
+
+result, error := collectors.CollectInformation("x86_64", MandatoryCollectors)
```

Comment:
Nice! I think that structure gives great flexibility wrt. implementing the collectors, and defining which collectors to use in which scenario. This opens up to have fine grained config options of which optional collectors to run for example.